### DISCUSSION TITLE
Remove a file from a bucket didn't work

### DISCUSSION THREAD

**Discussion by AurelliaChristie** [supabase/supabase#14512] [Questions] [answered]
Hey, 

I try to remove a file from supabase bucket using supabase-js client but it didn't work. There's no error return but the storage is not updated (i.e. the file isn't deleted). 
I've tried the solution from this discussion: ([#2466 ](https://github.com/orgs/supabase/discussions/2466)), but it still didn't work.

Here is my code:
`const { error } = await supabase.storage.from("images").remove([imgPath])`
where `imgPath` contains "project_images/file_name.png"`, which is the folder name & the file name of the file that I want to delete.

I also have added the required policy:
<img width="742" alt="image" src="https://github.com/supabase/supabase/assets/69672839/43a064b0-6291-4bd7-8a2d-77a29dfe8736">

- allow_delete: 
<img width="769" alt="image" src="https://github.com/supabase/supabase/assets/69672839/1e6094e4-8e05-4af7-8968-0e6bf43cc621">

- allow_select: 
<img width="765" alt="image" src="https://github.com/supabase/supabase/assets/69672839/99803cb2-a19d-450f-a988-a62ca5cde883">

Have tried to change the "role" for `allow_select` to be `anon` & also the "role" for `allow_delete` to be `null = public` but none of them works. Can anyone help me to figure out what I did wrong? Thank you

**Comment by GaryAustin1** [comment, marked as answer]
If both were set to public and true then RLS should not be a factor.  
Double check the pathname.
Are you using the storage UI to see if deleted or using it from the client?

**Reply by AurelliaChristie** [reply]
Hi, thanks for the answer. Turns out set both to public solve the issue. Thank you!
